web/app/js/components/ConfigureProfilesMsg.jsx

```diff
@@ -53,21 +65,37 @@ class ConfigureProfilesMsg extends React.Component {
   };
 
   handleClose = () => {
-    this.setState({ open: false });
+    let state = this.state;
+    state.error['namespace'] = false;
+    state.error['service'] = false;
+    state.open = false;
+    this.setState(state);
   };
 
   handleChange = name => {
     let state = this.state;
 
     return e => {
       state.query[name] = e.target.value;
+      state.error[name] = false;
       this.setState(state);
     };
-  }
+  };
+
+  validateFields = (type, name) => {
+    let match;
+    let error = this.state.error;
+    let serviceNameRegexp = RegExp(dns1123ServiceFmt);
```

Comment:
We can combine these and lines 44, as these regexps don't need to be redefined here on every validate call. So we can have something like:
```
const dns1035ServiceFmt = '^[a-z]([-a-z0-9]*[a-z0-9])?$';
const dns1123NamespaceFmt = '^[a-z0-9]([-a-z0-9]*[a-z0-9])?$';
const serviceNameRegexp = RegExp(dns1035ServiceFmt);
const namespaceNameRegexp = RegExp(dns1123NamespaceFmt);
```

Additionally, the ternary on line 91 can be simplified if we assign match in that line too:
```
  validateFields = (type, name) => {
    let error = this.state.error;
    let match = type === 'service' ?
      serviceNameRegexp.test(name) :
      namespaceNameRegexp.test(name);

    error[type] = !match;
    this.setState({ error });
  };
```